Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -290,129 +74,19 @@ extension ByteBuffer: IStreamable {
         case .begin:
             currentIndex = Int(offset)
         case .end:
-            currentIndex = array.count + Int(offset)
+            currentIndex = data.count + Int(offset)
         }
     }
 
     public func read(buffer: UnsafeMutableBufferPointer<UInt8>) -> Int? {
-        let endIndex = currentIndex + min(array.count - currentIndex, buffer.count)
+        let endIndex = currentIndex + min(data.count - currentIndex, buffer.count)
         guard currentIndex != endIndex else {
             return nil
         }
 
-        let dataArray = array[currentIndex..<endIndex]
+        let dataArray = data[currentIndex..<endIndex]
         dataArray.copyBytes(to: buffer, count: dataArray.count)
         currentIndex = endIndex
         return dataArray.count
     }
 }
-
-extension ByteBuffer {
-    /// initialize a  new `ByteBuffer` instance from `Foundation.Data`
-    public convenience init(data: Data) {
-        self.init(size: data.count)
-        put(data)
-    }
-
-    /// initialize a new `ByteBuffer` instance from `Foundation.InputStream`
-    public convenience init(stream: InputStream) throws {
-        self.init(size: 0)
-        stream.open()
-        defer {
-            stream.close()
-        }
-
-        let bufferSize = 1024
-        let buffer: UnsafeMutablePointer<UInt8> = allocatePointer(bufferSize)
-        defer {
-            buffer.deallocate()
-        }
-
-        while stream.hasBytesAvailable {
-            let read = stream.read(buffer, maxLength: bufferSize)
-            if read < 0 {
-                // Stream error occured
-                throw stream.streamError!
-            } else if read == 0 {
-                // EOF
-                break
-            }
-            allocate(read)
-            putByte(buffer.pointee)
-        }
-    }
```

Comment:
I didn’t find any usage in SDK. Also, this feels like a bad function which just loads the entire input stream to memory.